### PR TITLE
Add a method accepting std::string for filter_provides()

### PR DESCRIPTION
**Pull request by j-mracek**
This is required for Python binding, because when string argument is provided, binding convert it to vector of characters. The change might be problematic because filter_provides({"dnf"}) stop to work.

### CODE REVIEW

File: include/libdnf5/rpm/package_query.hpp [hunk 1]
```diff
@@ -281,6 +281,18 @@ class PackageQuery : public PackageSet {
     /// @since 5.0
     //
     // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_PROVIDES
+    void filter_provides(const std::string & pattern, libdnf5::sack::QueryCmp cmp_type = libdnf5::sack::QueryCmp::EQ) {
```
Comment (kontura) [inline include/libdnf5/rpm/package_query.hpp:284]: If we go with this version the docs need an update, it says `A vector of strings ..` but it takes just a string.

**Comment by j-mracek** [comment]
@jan-kolarik This PR does not change API,  but some project might got an issue with compilation. What do you suggest with this PR?

**Comment by jan-kolarik** [comment]
> @jan-kolarik This PR does not change API, but some project might got an issue with compilation. What do you suggest with this PR?

Hmm, could you please specify the exact issue with compiling the project after these changes are merged?

**Comment by jan-kolarik** [comment]
> The problem is described in commit message: The change might be problematic because filter_arch({"noarch"}) and filter_arch({"src", "nosrc"}) stop to work. It is a reason why commit [8492018](https://github.com/rpm-software-management/dnf5/commit/8492018084fcafe3d1c18f64d0eb1927667d6b2b) also modify additional lines (e.g. `query.filter_provides({"prv-all"});`). The additional overload method cause that compiler is unable to pick the correct methods because both std::string and std::vectorstd::string accept `char *`.

Thanks, I see now. Well, I feel that all changes that could result in breaking the existing client code should be delivered with the major version bump. Therefore, I'd like to bundle this together with other changes in https://github.com/rpm-software-management/dnf5/issues/1091. Is it OK like that?

**Comment by jan-kolarik** [comment]
@j-mracek Could you please target the PR against the https://github.com/rpm-software-management/dnf5/tree/dnf5-5.2.0.0 branch?

**Comment by j-mracek** [comment]
Opened - https://github.com/rpm-software-management/dnf5/pull/1216

**Pull request closed without merging**